core/src/HIP/Kokkos_HIP_Parallel_Team.hpp

```diff
@@ -433,6 +433,9 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
   int m_shmem_size;
   void* m_scratch_ptr[2];
   int m_scratch_size[2];
+  // Only let one ParallelFor/Reduce modify the team scratch memory. The
+  // constructor acquires the mutex which is released in the destructor.
+  std::unique_lock<std::mutex> m_scratch_lock;
```

Comment:
`std::lock_guard` is non-copyable.  This means you implicitly deleted the copy constructor and copy assignment.  Was it intentional?  Did you make sure it plays well with the kernel launching?

Reply:
>  This means you implicitly deleted the copy constructor and copy assignment.

True at the same time if you use the copy constructor currently, you are copying pointers without copying their data... Also I don't see what's your use case for that. Also I am using `std::unique_lock` instead of `std::lock_guard`. Unlike `std::lock_guard`, `std::unique_lock` is movable.

>  Did you make sure it plays well with the kernel launching?

I am not sure what you mean by that but all the tests pass on Tulip and on the CI.

Reply:
I missed it was `std::unique_lock` and not `std::lock_guard`.  I can't remember how we pass the driver around in the kernel launching.  This is not a trick question I was just curious if you considered it.